### PR TITLE
transition from ansi-wl-pprint to prettyprinter

### PR DESCRIPTION
**Pull request by Ptival**
Prep work for transitioning away from `ansi-wl-pprint`.

Will likely need some synchronized changes in dependencies.

### CODE REVIEW

File: src/Flexdis86/InstructionSet.hs [hunk 1]
```diff
@@ -36,14 +38,14 @@ import           Flexdis86.Relocation
...
 import           Flexdis86.Sizes
 
-padToWidth :: Int -> String -> String
-padToWidth n s = if l > 0 then s ++ (replicate (n - l) ' ') else s
+padToWidth :: Int -> String -> PP.Doc a
+padToWidth n s = fromString $ if l > 0 then s ++ replicate (n - l) ' ' else s
   where l = length s
 
-ppPunctuate :: Doc -> [Doc] -> Doc
+ppPunctuate :: PP.Doc a -> [PP.Doc a] -> PP.Doc a
```
Comment (kquick) [inline src/Flexdis86/InstructionSet.hs:45]: This is just `PP.hsep . PP.punctuate`

File: src/Flexdis86/Register.hs [hunk 2]
```diff
@@ -124,7 +126,9 @@ pattern HighReg8 w <- (asHighReg -> Just w)
   where HighReg8 x = high_reg8 x
 
 instance Show Reg8 where
-  show (Reg8 i) = assert (i < 20) (regNames8 V.! (fromIntegral i))
+  show (Reg8 i) = assert (i < 20) (regNames8 V.! fromIntegral i)
+
+instance PP.Pretty Reg8 where pretty = fromString . show
```
Comment (kquick) [inline src/Flexdis86/Register.hs:131]: `instance PP.Pretty Reg8 where pretty = PP.unsafeViaShow`

Comment (kquick) [reply src/Flexdis86/Register.hs:131]: and the 8 others following.

Comment (kquick) [reply src/Flexdis86/Register.hs:131]: and the next 2 files

**Review by kquick — APPROVED** [review]
That was fast!  A couple of recommendations for more `prettyprinter`-aligned changes, but generally looks good.

My main concern would be if we have round-trip tests that could help ensure this doesn't introduce any unintended changes in the output?

**Comment by Ptival** [comment]
I did a comparative test over a binary with ~5k individual instructions, and found no discrepancy after my last patch.

**Comment by kquick** [comment]
> I did a comparative test over a binary with ~5k individual instructions, and found no discrepancy after my last patch.

Thanks!  I almost prefer the spaces after the commas, but deviation like that would be better left for a different effort.  This looks good to me.

**Comment by RyanGlScott** [comment]
I'm curious if there are any breakages that are uncovered when trying to use this change to build the `macaw-x86-symbolic`. (If you can get that far, there's a pretty good chance that SAW will also work.)

**Comment by Ptival** [comment]
All of `macaw` (i.e. `cabal build all`) seems to build with these changes.

**Review by RyanGlScott — APPROVED** [review]
Fantastic. If both `macaw-x86` and `reopt` work, then I think that covers the vast majority of code that would be impacted.

**Comment by Ptival** [comment]
Alright, I'll attempt a merge of this, and I'll take responsibility if it affects anyone downstream.

**Pull request merged**